src/plugins/iam/sagas/index.js

```diff
@@ -10,15 +10,20 @@ import { actionTypes, serviceIds } from '../constants';
  * @param {Container} container
  */
 function* watchLogin(container) {
+  let lastTask;
   while (true) {
     const { doLogin } = yield race({
       doLogin: take(actionTypes.LOGIN_REQUESTED),
       doLogout: take(uiActionTypes.LOGOUT_REQUESTED),
     });
 
+    if (lastTask) {
```

Comment:
@hunyoboy might call this `loginTask` and only check to see if we need to cancel it in the `if (doLogin)`block

Reply:
@efjacobson thank you i got you but i think we also need to cancel if for some reason a logout request will be issued. if for some reason the login flow is still running ( auth server still pending, etc. ) and suddenly we want to logout, i think we really need to terminate the login flow to keep the state in sync.